### ISSUE TITLE
Folder upload fails due to broken order of actions

### ISSUE THREAD

**Issue by LukasHirt** [owncloud/phoenix#3351]
Folder upload sometimes fails because of parent folders not being created before the files are being uploaded.

Currently, we use two different methods for uploading folder:

one in "New" dialogue

https://github.com/owncloud/phoenix/blob/06554bef6ae1881b7646e7d126dbe36ad6e85a66/apps/files/src/mixins.js#L263

the second one in drag and drop

https://github.com/owncloud/phoenix/blob/06554bef6ae1881b7646e7d126dbe36ad6e85a66/apps/files/src/mixins.js#L177

Let's try to unite them.

**Comment by pascalwengerter** [comment]
@LukasHirt is this still the case? :)

**Comment by LukasHirt** [comment]
Hmm, need to test that but folder upload hasn't been touched so I don't think it would be solved 🤷

**Comment by LukasHirt** [comment]
Cannot reproduce the bug but I would still keep the ticket for the research if we can unite folder upload dialog and drag and drop underlying logic...

**Comment by pascalwengerter** [comment]
> Cannot reproduce the bug but I would still keep the ticket for the research if we can unite folder upload dialog and drag and drop underlying logic...

Good idea! Closed as refactoring the upload logic is tracked in the more pressing #5371, linking here for reference
